frontend/src/metabase/dashboard/components/DashboardLeaveConfirmationModal/utils.ts

```diff
@@ -0,0 +1,24 @@
+import type { Location } from "history";
+
+import { deserializeCard, parseHash } from "metabase/query_builder/actions";
+
+export const isNavigatingToCreateADashboardQuestion = (
```

Comment:
- I should add some tests for this
- `location` should be named `nextLocation` for clarify